compiler-core/src/config.rs

```diff
@@ -173,6 +177,29 @@ impl PackageConfig {
         .expect("internal module globs")
         .is_match(module)
     }
+
+    // Checks to see if the gleam version specified in the config is compatable
+    // with the current compiler version
+    pub fn check_gleam_compatability(&self) -> Result<(), Error> {
+        if let Some(required_version) = &self.gleam_version {
+            let compiler_version = hexpm::version::Version::parse(env!("CARGO_PKG_VERSION"))
```

Comment:
Let's use a constant for the Gleam version. Currently it's in `compiler-cli/src/main.rs`, let's move it to `compiler-core/src/lib.rs`.